### PR TITLE
Support/live 2473 sign message

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### 📝 Description

Add signMessage call management in the mobile app.
Modification of the current management of the `onClose` callback of the `HeaderRightClose.js` component. This callback was not without some potential side effect. Therefore, its call has been moved at the end of the execution of the `HeaderRightClose::close` method.
The WalletConnect integration for the mobile has been slightly modified, due to this change.

This is the new version of [previous PR](https://github.com/LedgerHQ/ledger-live/pull/651) on the same ticket.

### ❓ Context

- **Impacted projects**: `Mobile`, `WalletConnect`
- **Linked resource(s)**: 
  - [LIVE-2473](https://ledgerhq.atlassian.net/browse/LIVE-2473)
  - [EPIC](https://ledgerhq.atlassian.net/browse/LIVE-2463) related
  - Other PR related
    - https://github.com/LedgerHQ/live-app-sdk/pull/118
    - https://github.com/LedgerHQ/eth-dapp-browser/pull/9
    - https://github.com/LedgerHQ/platform-app-debug/pull/3

### ✅ Checklist

- [ ] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

No current behavior of the mobile application is expected with this sole PR.

### CODE REVIEW

**Review by chabroA — APPROVED** [review]

File: apps/ledger-live-mobile/src/components/HeaderRightClose.js [hunk 1]
```diff
@@ -34,22 +34,30 @@ export default function HeaderRightClose({
   const [onModalHide, setOnModalHide] = useState();
 
   function close(): void {
-    if (onClose) {
-      onClose();
-    }
-
     if (skipNavigation) {
+      // onClose should always be call at the end of the close method,
```
Comment (hedi-edelbloute) [inline apps/ledger-live-mobile/src/components/HeaderRightClose.js:38]: Nitpicking but I think it's : `onClose should always be called at the end of the close method`

Comment (chabroA) [inline apps/ledger-live-mobile/src/components/HeaderRightClose.js:38]:
```suggestion
      // onClose should always be called at the end of the close method,
```

Comment (chabroA) [inline apps/ledger-live-mobile/src/components/HeaderRightClose.js:38]:
```suggestion
      // onClose should always be called at the end of the close method,
```

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js [hunk 2]
```diff
@@ -542,6 +547,40 @@ const WebPlatformPlayer = ({ manifest, inputs }: Props) => {
...
+  const signMessage = useCallback(
+    ({ accountId, message }: { accountId: string, message: string }) => {
+      let formattedMessage: MessageData | null;
+
+      try {
+        const account = accounts.find(account => account.id === accountId);
+        invariant(account, "account not found");
+        formattedMessage = prepareMessageToSign(account, message);
+      } catch (error) {
+        return Promise.reject(error);
```
Comment (Justkant) [inline apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js:559]: I think we're just missing the tracking for success and failure

**Review by Justkant — CHANGES_REQUESTED** [review]
We just need to add the tracking

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js (hunk 2 above)
Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js:559]: @Justkant I don't find the info to put to the tracking for this US. Do you what info should be added?

File: apps/ledger-live-mobile/src/components/HeaderRightClose.js [hunk 3]
```diff
@@ -34,22 +34,30 @@ export default function HeaderRightClose({
...
-    if (onClose) {
-      onClose();
-    }
-
     if (skipNavigation) {
+      // onClose should always be called at the end of the close method,
+      // so the callback will not interfere with the expected behavior of this component
+      if (onClose) {
+        onClose();
+      }
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/components/HeaderRightClose.js:42]: @Justkant , I don't see any obvious place to write this function. Do you have a suggestion in `common` package?

File: apps/ledger-live-mobile/src/components/HeaderRightClose.js [hunk 4]
```diff
@@ -34,22 +34,30 @@ export default function HeaderRightClose({
   const [onModalHide, setOnModalHide] = useState();
 
   function close(): void {
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/components/HeaderRightClose.js:36]: Is this a common best practice in Ledger?

File: apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js (hunk 2 above)
Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/components/WebPlatformPlayer/index.js:560]: Just added the method by following the same implement as the other tracking methods.

File: apps/ledger-live-mobile/src/components/HeaderRightClose.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply apps/ledger-live-mobile/src/components/HeaderRightClose.js]: In the meantime add a default implementation for the `onClose` method.
I think the best would be to replace in every call for this 'emptyFunction'.

File: apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/index.jsx [hunk 5]
```diff
@@ -416,30 +416,33 @@ const WebPlatformPlayer = ({ manifest, onClose, inputs, config }: Props) => {
       try {
         formattedMessage = prepareMessageToSign(account, message);
       } catch (error) {
+        tracking.platformSignMessageFail(manifest);
         return Promise.reject(error);
       }
 
+      tracking.platformSignMessageRequested(manifest);
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/index.jsx:423]: IMO we should just put the requested tracking before the first fail

**Review by Justkant — CHANGES_REQUESTED** [review]

**Review by Justkant — APPROVED** [review]

**Pull request merged**